spec/blocksync/verification.md

```diff
@@ -0,0 +1,116 @@
+## Block verification
+
+When blocksyncing, a node is not participating in conensus. It is receiving blocks that have already been decided and committed. To avoid being fooled by malicious peers, the node has to verify the received blocks before executing the transactions and storing the block in its store. 
+
+
+The verification in blocksync aims to apply the same logic as the [light client verification](../light-client/verification/README.md). The model relies on the existence of an **initial trusted state**. 
+
+Based on this state we verify subsequent blocks. 
+
+### Trusted state
+
+The light client additionally relies on the notion of a **trusting period**. A trusting period is the time during which we assume we can trust validators because, if we do detect misbehaviour, we can slash them - they are still bonded. Beyond this period, the validators do not have to have any bonded assets and cannot be held accountable for their misbheaviour. Blocksync-ing blocks will most often be outside this trusting period for a particular block. Therefore, the trusting period assumptions as they are in the light client, cannot be applied here. 
```

Comment:
This is a bit unclear to me. In my view, the trusting period is nothing specific to the light client, but a consequence of the unbonding period. If we cannot apply the trusting period, this means that the whole protocol is in general unsafe, or more precisely, the problem we want to solve is unsolvable. 

For instance, if we want to blocksync from genesis, and genesis is 2 months old, all the initial validators might have unbonded. They can produce an alternative history, and make our new node always sync to a bad state. If we don't make additional assumptions the problem is unsolvable. (Or the protocol is just best-effort. But even then we should explain in what cases it works).

This means to say something meaningful about blocksync, we might need to add (or make explicit already existing implicit) assumptions, e.g., every new node can talk to at least one correct peer who has the correct history of blocks (headers) stored.

Reply:
The main point I agree with you on is the weak guarantees this initial trust in the validator set provides to us in blocksync. And I think we need to find a way to improve the guarantees on this by either starting a light client service to verify headers against or some other, safer solution.

I think there is a sentence about the node always having a correct peer but correct is not specified fully as you stated it here. Indeed, for this to be correct we do need  a peer who will send us a correct block (header), but I feel this assumption is weak. I will add this assumption explicitly for now but I do think we should come up with a safer solution.

Reply:
One thing that may inform the assumptions is the use case. Perhaps we can differentiate:
- If you have a trusted state within the trusting period (w.r.t. to now): strong guarantees
- Otherwise: weak guarantees. We still provide automation, but you should double check the outcome.

For me the idea with witnesses is in the second option. "We cannot really give guarantees, but we want to make the attack more complicated." 

We should make the different use cases/assumptions and the following guarantees more clear.

Reply:
If we launched the light client as a service (as state sync does), do you think we could always provide strong guarantees? if so would this be a better option compared to witness verification? If we are satisfied with the witness model then I can certainly specify these things better.